checkstyle/src/files_matching_glob_patterns.rs

```diff
@@ -0,0 +1,58 @@
+use glob::Pattern;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+fn list_files_in_folder(folder_path: &Path) -> Result<Vec<PathBuf>, std::io::Error> {
+    let mut file_paths = Vec::new();
+
+    for entry in fs::read_dir(folder_path)? {
+        let entry = entry?;
+        let path = entry.path();
+
+        if path.is_symlink() {
+            continue;
+        }
+
+        if path.is_file() {
+            file_paths.push(path.clone());
+        } else if path.is_dir() {
+            if let Ok(sub_paths) = list_files_in_folder(&path) {
+                file_paths.extend(sub_paths);
+            }
+        }
+    }
+
+    Ok(file_paths)
+}
+
+fn all_files(folder_path: &Path) -> Vec<PathBuf> {
+    list_files_in_folder(folder_path).unwrap_or_default()
+}
+
+pub fn files_matching_patterns(

```

Comment:
The full path of this public function is `files_matching_glob_patterns::files_matching_patterns`. This is quite verbose and duplicates info. 

How about naming this module `glob` and this function `matching_files`? Together, we'd get the full path `glob::matching_files`, which is nicer in my opinion.